Hyperbolic_surface_triangulation_2/include/CGAL/Hyperbolic_fundamental_domain_2.h

```diff
@@ -0,0 +1,210 @@
+  Hyperbolic_isometry_2<Traits> side_pairing(int index) const;// Returns the isometry that maps side A to side B, where B is the index-th side, and A is the side paired to B
+
+  void from_stream(std::istream& s);
+  void to_stream(std::ostream& s) const;
+
+  bool is_valid() const;
+
+private:
+  std::vector<Point>                                               _vertices;
+  std::vector<int>                                                  _pairings;
+};
+
+template<class Traits> std::ostream& operator<<(std::ostream& s, const Hyperbolic_fundamental_domain_2<Traits>& domain);
+template<class Traits> void operator>>(std::istream& s, Hyperbolic_fundamental_domain_2<Traits>& domain);
+
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+template<class Traits>
+Hyperbolic_fundamental_domain_2<Traits>::Hyperbolic_fundamental_domain_2() {}
+
+template<class Traits>
+Hyperbolic_fundamental_domain_2<Traits>::Hyperbolic_fundamental_domain_2(const std::vector<Point>& vertices, const std::vector<int>& pairings){
+  set(vertices, pairings);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+template<class Traits>
+void Hyperbolic_fundamental_domain_2<Traits>::set(const std::vector<Point>& vertices, const std::vector<int>& pairings){
+  _vertices = vertices;
+  _pairings = pairings;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+template<class Traits>
+int Hyperbolic_fundamental_domain_2<Traits>::size() const{
+  return _vertices.size();
+}
+
+template<class Traits>
+typename Hyperbolic_fundamental_domain_2<Traits>::Point Hyperbolic_fundamental_domain_2<Traits>::vertex(int index) const{
+  return _vertices[index];
+}
+
+template<class Traits>
+int Hyperbolic_fundamental_domain_2<Traits>::paired_side(int index) const{
+  return _pairings[index];
+}
+
+template<class Traits>
+Hyperbolic_isometry_2<Traits> Hyperbolic_fundamental_domain_2<Traits>::side_pairing(int index) const{
+  int n = size();
+  int paired_index = paired_side(index);
+
+  Point p1,p2,q1,q2;
+  q1 = vertex(index);
+  q2 = vertex((index+1)%n);
+  p2 = vertex(paired_index);
+  p1 = vertex((paired_index+1)%n);
+
+  Hyperbolic_isometry_2<Traits> isom = isometry_pairing_the_sides<Traits>(p1,p2,q1,q2);
+  return isom;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+template<class Traits>
+void Hyperbolic_fundamental_domain_2<Traits>::to_stream(std::ostream& s) const{
+  int n = size();
+
+  s << std::to_string(n) << std::endl;
+
+  for (int k=0; k<n; k++){
+    s << paired_side(k) << std::endl;
+  }
+
+  for (int k=0; k<n; k++){
+    s << vertex(k) << std::endl;
+  }
+}
```

Comment:
Instead of `void`, return the stream:
```suggestion
std::ostream& Hyperbolic_fundamental_domain_2<Traits>::to_stream(std::ostream& s) const{
  int n = size();

  s << std::to_string(n) << std::endl;

  for (int k=0; k<n; k++){
    s << paired_side(k) << std::endl;
  }

  for (int k=0; k<n; k++){
    s << vertex(k) << std::endl;
  }
  return s;
}
```

Reply:
done